src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -264,12 +264,12 @@ def _before_record_request(request):
         request.uri = re.sub('/sig=([^/]+)&', '/sig=0000&', request.uri)
         request.uri = _scrub_deployment_name(request.uri)
         # remove randomized portion from resource group name
-        resource_group_name = re.search('resource[Gg]roups/([^/]+_)/', request.uri)
+        resource_group_name = re.search('resource[Gg]roups/([^/^?]+_)', request.uri)
```

Comment:
@tjprescott, the existing regular expression doesn't handle the case when the resource group name is the last portion of url path, particularly when you want to `get` a resource group to use it's location, which is a common case. Here I am fixing it.
However, we might still have an issue that since we don't fix the `name` or `id` in the body, and this will fail related verification. Thoughts?